WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -274,6 +278,7 @@ private async Task StepTransactionSigningPhaseAsync(CancellationToken cancellati
 				if (state.IsFullySigned)
 				{
 					Transaction coinjoin = state.CreateTransaction();
+					CoinjoinTransactionCreated?.Invoke(this, new CoinjoinTransactionCreatedEventArgs(round.Id, coinjoin));
```

Comment:
I'm not sure If I understand. Is this what you mean?
```
	private Transaction CreateTransaction(Round round)
	{
		SigningState state = round.Assert<SigningState>();
		Transaction coinjoin = state.CreateTransaction();
		CoinjoinTransactionCreated?.SafeInvoke(this, new CoinjoinTransactionCreatedEventArgs(round.Id, coinjoin));
		return coinjoin;
	}
```

> Also, I would use SafeInvoke everywhere.

Done in https://github.com/zkSNACKs/WalletWasabi/pull/9769/commits/3ea9339b9796b5a1735e024ce0083d8d272e1b16.

Reply:
Yes, something like that. Perhaps I would pass the SigningState instead of the Round but you got the idea.

Reply:
> Perhaps I would pass the SigningState instead of the Round but you got the idea.

That's not possible because the method has to know round id.

Which one do you prefer?
```c#
private Transaction CreateTransaction(Round round)
```
```c#
private Transaction CreateTransaction(uint256 roundId, SigningState signingState)
```

Reply:
You're right. Both are okay to me but the second one is more correct to me.

Reply:
Done in https://github.com/zkSNACKs/WalletWasabi/pull/9769/commits/e31340c4250f3a129cb0c46102a155550d493ba7.